src/wallet/spend.cpp

```diff
@@ -959,20 +960,7 @@ static util::Result<CreatedTransactionResult> CreateTransactionInternal(
         return util::Error{_("Missing solving data for estimating transaction size")};
     }
     CAmount fee_needed = coin_selection_params.m_effective_feerate.GetFee(nBytes);
-    CAmount current_fee = result->GetSelectedValue() - recipients_sum - change_amount;
-
-    // The only time that fee_needed should be less than the amount available for fees is when
-    // we are subtracting the fee from the outputs. If this occurs at any other time, it is a bug.
-    if (!coin_selection_params.m_subtract_fee_outputs && fee_needed > current_fee) {
-        return util::Error{Untranslated(STR_INTERNAL_BUG("Fee needed > fee paid"))};
-    }
-
-    // If there is a change output and we overpay the fees then increase the change to match the fee needed
-    if (nChangePosInOut != -1 && fee_needed < current_fee) {
-        auto& change = txNew.vout.at(nChangePosInOut);
-        change.nValue += current_fee - fee_needed;
-        current_fee = fee_needed;
-    }
+    CAmount current_fee = result->GetSelectedValue() - CalculateOutputValue(txNew);
```

Comment:
In 1574bfa0:

Hmm, guess that the `CalculateOutputValue` is meant to be a guard? Because, atm, there seems to not be possible to have `recipients_sum + change_amount != CalculateOutputValue(txNew)`

If that is the case, what about adding an `Assume` here?

Reply:
Done